src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.cs

```diff
@@ -134,12 +134,31 @@ protected void SkipToAndParseCode(Func<HtmlSymbol, bool> condition)
                 {
                     if (last != null)
                     {
+                        if (startOfLine && last.Type == HtmlSymbolType.WhiteSpace)
+                        {
+                            Output(SpanKind.Markup);
+                            Span.ChunkGenerator = SpanChunkGenerator.Null;
+                        }
+
                         Accept(last);
                         last = null;
                     }
+
                     AddMarkerSymbolIfNecessary();
                     Output(SpanKind.Markup);
+
                     RazorComment();
+
+                    // Handle the whitespace and newline at the end of a razor comment.
```

Comment:
Why does this only consume the whitespace if `startOfLine` is `true`? Shouldn't the final newline be removed in (less common I'm sure but possible) cases like `@* hello *@      @* world *@`?